### PR TITLE
[KOGITO-4957] - Handling HTTP CloudEvents requests without Content-Type

### PR DESCRIPTION
**Pull request by ricardozanini**
Many thanks for submitting your Pull Request :heart:! 

In this PR, we reevaluate the "Consumes" header of the CloudEvent HTTP Listener endpoint. Now, we do accept everything and try to parse the data field as JSON internally. This is needed because a client might send an HTTP request without specifying a Content-Type. In this case, the CE SDK will treat it as a BINARY format, so the data will be serialized to base64 data by Jackson.

To avoid this, in situations like this, we try to parse the data to JSON before pushing the message to internal channels. Then, if it's not a JSON message, we fail beautifully.

See: https://issues.redhat.com/browse/KOGITO-4957 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/master/kogito-ide-config)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Comment by ricardozanini** [comment]
OH MY GOD, IT'S GREEN!

Please don't make me do any modifications to this PR. :laughing:

**Comment by mouachan** [comment]
Hey,
Tested with the reproducer and works fine :)

cheers

**Review by cristianonicolai — APPROVED** [review]

File: kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/events/CloudEventsResourceGenerator.java [hunk 1]
```diff
@@ -15,46 +15,26 @@
...
 import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
-import com.github.javaparser.ast.body.FieldDeclaration;
 
 public class CloudEventsResourceGenerator extends AbstractEventResourceGenerator {
 
-    static final String EMITTER_PREFIX = "emitter_";
     private static final String CLASS_NAME = "CloudEventListenerResource";
 
-    private final KogitoBuildContext context;
-    private final List<TriggerMetaData> triggers;
```
Comment (danielezonca) [inline kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/events/CloudEventsResourceGenerator.java:40]: Aren't these metadata required for kogito operator integration?

**Review by danielezonca — APPROVED** [review]
Approving :)

File: kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/events/CloudEventsResourceGenerator.java (hunk 1 above)
Comment (ricardozanini) [reply kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/events/CloudEventsResourceGenerator.java:40]: This info is generated by the `TopicInformationResource`: https://github.com/kiegroup/kogito-runtimes/blob/master/addons/cloudevents/cloudevents-common-addon/src/main/java/org/kie/kogito/addon/cloudevents/AbstractTopicDiscovery.java#L39

File: addons/cloudevents/cloudevents-quarkus-addon/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/http/CloudEventInvalidDataException.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.kogito.addon.cloudevents.quarkus.http;
+
+import io.cloudevents.CloudEvent;
+
+public class CloudEventInvalidDataException extends Exception {
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply addons/cloudevents/cloudevents-quarkus-addon/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/http/CloudEventInvalidDataException.java:20]: 'Cause we must catch it

**Pull request merged**